internal/opts/config.go

```diff
@@ -0,0 +1,192 @@
+package opts
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"time"
+
+	flag "github.com/spf13/pflag"
+
+	"github.com/spf13/viper"
+)
+
+type Config struct {
+
+	// region level configuration.
+	DisklessMode           bool   `mapstructure:"diskless"  desc:"Enables badger diskless mode where data is stored entirely in memory. "`
+	NodeName               string `mapstructure:"node-name" desc:"Node Name"`
+	DbEngine               string `mapstructure:"db-engine" desc:"Underlying DB engine for storing data - badger|rocksdb"`
+	DbEngineIni            string `mapstructure:"db-engine-ini" desc:"An .ini file for configuring the underlying storage engine. Refer badger.ini or rocks.ini for more details."`
+	DbRole                 string `mapstructure:"role" desc:"Role of the node - master|slave|standalone"`
+	ReplPollIntervalString string `mapstructure:"repl-poll-interval" desc:"Interval used for polling changes from master. Eg., 10s, 5ms, 2h, etc."`
+	BlockCacheSize         uint64 `mapstructure:"block-cache-size" desc:"Amount of cache (in bytes) to set aside for data blocks. A value of 0 disables block caching altogether."`
+	DcID                   string `mapstructure:"dc-id" desc:"DC / Availability zone identifier"`
+	Database               string `mapstructure:"database" desc:"Database identifier"`
+	VBucket                string `mapstructure:"vbucket" desc:"vBucket identifier"`
+
+	// Storage Configuration
+	RootFolder string `mapstructure:"root-folder" desc:"Root Dir (optional)"` // used to derive other folders if not defined
+	DbFolder   string `mapstructure:"db-folder" desc:"DB folder path for storing data files"`
+
+	// Server Configuration
+	ListenAddr string `mapstructure:"listen-addr" desc:"Address on which the DKV service binds"`
+	StatsdAddr string `mapstructure:"statsd-addr" desc:"StatsD service address in host:port format"`
+
+	//Service discovery related params
+	DiscoveryServiceConfig string `mapstructure:"discovery-service-config" desc:"A .ini file for configuring discovery service parameters"`
+
+	// Temporary variables to be removed once https://github.com/flipkart-incubator/dkv/issues/82 is fixed
+	// The above issue causes replication issues during master switch due to inconsistent change numbers
+	// Thus enabling hardcoded masters to not degrade current behaviour
+	ReplicationMasterAddr string `mapstructure:"repl-master-addr" desc:"Service address of DKV master node for replication"`
+	DisableAutoMasterDisc bool   `mapstructure:"disable-auto-master-disc"`
```

Comment:
Let's change this to an affirmative variable. `DisableAutoMasterDisc ` -> `EnableAutoMasterDisc`. Default: `true`

Reply:
This should be a separate PR, since this changes the logic. Also both of these flags could be eventually also removed with autodiscovery taking care of the entire thing.

Reply:
Created https://github.com/flipkart-incubator/dkv/issues/128